### PR TITLE
MU WPCOM: Move the Event Countdown and Timeline blocks from jetpack to jetpack-mu-wpcom

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/8036#issuecomment-2222306034

## Proposed changes:

* Move the following blocks from jetpack to jetpack-mu-wpcom
  * jetpack/event-countdown
  * jetpack/timeline

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply changes to your site
* Go to the editor
* Insert the Event Countdown block
* Make sure it works as before
* Insert the Timeline block
* Make sure it works as before

### CODE REVIEW

**Review by taipeicoder — APPROVED** [review]
Tested and LGTM 👍

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-blocks/timeline/timeline.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	return $content;
+}
+
+/**
+ * Load assets on the editor.
+ */
+function enqueue_block_editor_assets() {
+	\jetpack_mu_wpcom_enqueue_assets( 'wpcom-blocks-timeline-editor', array( 'js', 'css' ) );
+}
+if ( is_admin() ) {
```
Comment (ellatrix) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-blocks/timeline/timeline.php:54]: Why did you add these conditions? It should always be `enqueue_block_editor_assets` for block scripts that register the blocks. I don't see it present before this commit. See #41055.
